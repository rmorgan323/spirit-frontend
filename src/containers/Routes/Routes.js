/*eslint-disable no-useless-escape*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import * as actions from '../../actions';

import authLogin from '../../helpers/authLogin';
import PatientDashboard from '../PatientDashboard/PatientDashboard';
import Sensory from '../../components/Sensory/Sensory';
import Modulation from '../../components/Modulation/Modulation';
import Executive from '../../components/Executive/Executive';
import Postural from '../../components/Postural/Postural';
import Social from '../../components/Social/Social.js';
import Spirit from '../../components/Spirit/Spirit.js';
import Join from '../Join/Join';
import Create from '../Create/Create';
import UserDashboard from '../UserDashboard/UserDashboard';
import clearLocalStorage from '../../helpers/clearLocalStorage/clearLocalStorage';
import Users from '../Users/Users';
import Header from '../Header/Header';
import Sessions from '../Sessions/Sessions';
import PatientHeader from '../PatientHeader/PatientHeader';
import ConcernHeader from '../ConcernHeader/ConcernHeader';
import Sam from '../../components/Sam/Sam';
import SessionTabs from '../SessionTabs/SessionTabs';
import ViewSession from '../../containers/ViewSession/ViewSession';
import Finish from '../Finish/Finish';

export class Routes extends Component {
  componentDidMount = async () => {
    await this.checkForKey();
    await this.props.getUser();
    await this.props.getDefinitions();
  };

  checkForKey = () => {
    const key = localStorage.getItem('spirit-token-987');
    if (window.location.search.length > 200) {
      const token = this.getToken();
      localStorage.setItem('spirit-token-987', token);
    } else if (key) {
      return;
    } else {
      clearLocalStorage();
      window.location = authLogin;
      return;
    }
  };

  getToken = () => {
    const removeFront = new RegExp(/\?token=/);
    const removeBack = new RegExp(/\&signup=true/);
    const parse = window.location.search.replace(removeFront, '');
    const parseEnd = parse.replace(removeBack, '');

    return parseEnd;
  };

  render() {
    return (
      <div>
        <Route path="/" component={Header} />
        <Route path="/spirit/patients" component={PatientHeader} />
        <Route path="/spirit/concerns" component={PatientHeader} />
        <Route path="/spirit/sessions" component={PatientHeader} />

        <Route path="/spirit/concerns" component={ConcernHeader} />
        <Route path="/spirit/sessions" component={ConcernHeader} />
        <Route path="/spirit/sessions/:session_id" component={SessionTabs} />

        <Switch>
          <Route path="/spirit/sessions/:session_id/sam" component={Sam} />
          <Route
            path="/spirit/sessions/:session_id/sensory"
            component={Sensory}
          />

          <Route
            path="/spirit/sessions/:session_id/modulation"
            component={Modulation}
          />

          <Route
            path="/spirit/sessions/:session_id/executive"
            component={Executive}
          />

          <Route
            path="/spirit/sessions/:session_id/postural"
            component={Postural}
          />

          <Route
            path="/spirit/sessions/:session_id/social"
            component={Social}
          />

          <Route
            path="/spirit/sessions/:session_id/finish"
            component={Finish}
          />

          <Route 
            path="/spirit/sessions/:session_id/view"
            component={ViewSession}
          />

          <Route
            path="/spirit/concerns/:concernId/sessions"
            component={Sessions}
          />

          <Route
            path="/spirit/patients/:patient_id"
            component={PatientDashboard}
          />

          <Route exact path="/spirit/users/:user_id/join" component={Join} />

          <Route
            exact
            path="/spirit/users/:user_id/create"
            component={Create}
          />

          <Route
            exact
            path="/spirit/users/:user_id"
            component={UserDashboard}
          />

          <Route path="/spirit/users" component={Users} />

          <Route path="/spirit/index" component={Spirit} />
        </Switch>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  getUser: () => {
    dispatch(actions.getUser());
  },
  getDefinitions: () => {
    dispatch(actions.getDefinitions());
  }
});

export default connect(null, mapDispatchToProps)(Routes);

Routes.propTypes = {
  getUser: PropTypes.func,
  getDefinitions: PropTypes.func
};
