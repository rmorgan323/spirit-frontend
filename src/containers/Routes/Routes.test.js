import React from 'react';
import { Routes, mapDispatchToProps } from './Routes';
import { shallow } from 'enzyme';

describe('Routes tests', () => {
  let getUser;
  let getDefinitions;
  let renderedRoutes;

  beforeEach(() => {
    getUser = jest.fn();
    getDefinitions = jest.fn();

    renderedRoutes = shallow(
      <Routes getUser={getUser} getDefinitions={getDefinitions} />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedRoutes).toMatchSnapshot();
  });
});

describe('mapDispatchToProps tests', () => {
  it('Should call dispatch when getUser is called', () => {
    const mockDispatch = jest.fn();
    const result = mapDispatchToProps(mockDispatch);

    result.getUser();
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('Should call dispatch when getDefinitions is called', () => {
    const mockDispatch = jest.fn();
    const result = mapDispatchToProps(mockDispatch);

    result.getDefinitions();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
