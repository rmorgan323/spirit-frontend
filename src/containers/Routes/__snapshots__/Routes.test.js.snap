// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes
    getDefinitions={[Function]}
    getUser={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          path="/"
        />,
        <Route
          component={[Function]}
          path="/spirit/patients"
        />,
        <Route
          component={[Function]}
          path="/spirit/concerns"
        />,
        <Route
          component={[Function]}
          path="/spirit/sessions"
        />,
        <Route
          component={[Function]}
          path="/spirit/concerns"
        />,
        <Route
          component={[Function]}
          path="/spirit/sessions"
        />,
        <Route
          component={[Function]}
          path="/spirit/sessions/:session_id"
        />,
        <Switch>
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/sam"
          />
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/sensory"
          />
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/modulation"
          />
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/executive"
          />
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/postural"
          />
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/social"
          />
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/finish"
          />
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id/view"
          />
          <Route
            component={[Function]}
            path="/spirit/concerns/:concernId/sessions"
          />
          <Route
            component={[Function]}
            path="/spirit/patients/:patient_id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/spirit/users/:user_id/join"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/spirit/users/:user_id/create"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/spirit/users/:user_id"
          />
          <Route
            component={[Function]}
            path="/spirit/users"
          />
          <Route
            component={[Function]}
            path="/spirit/index"
          />
        </Switch>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/spirit/patients",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/spirit/concerns",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/spirit/sessions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/spirit/concerns",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/spirit/sessions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/spirit/sessions/:session_id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/sam"
            />,
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/sensory"
            />,
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/modulation"
            />,
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/executive"
            />,
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/postural"
            />,
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/social"
            />,
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/finish"
            />,
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/view"
            />,
            <Route
              component={[Function]}
              path="/spirit/concerns/:concernId/sessions"
            />,
            <Route
              component={[Function]}
              path="/spirit/patients/:patient_id"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/spirit/users/:user_id/join"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/spirit/users/:user_id/create"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/spirit/users/:user_id"
            />,
            <Route
              component={[Function]}
              path="/spirit/users"
            />,
            <Route
              component={[Function]}
              path="/spirit/index"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/sam",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/sensory",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/modulation",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/executive",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/postural",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/social",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/finish",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/sessions/:session_id/view",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/concerns/:concernId/sessions",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/patients/:patient_id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/spirit/users/:user_id/join",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/spirit/users/:user_id/create",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/spirit/users/:user_id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/users",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/spirit/index",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/spirit/patients"
          />,
          <Route
            component={[Function]}
            path="/spirit/concerns"
          />,
          <Route
            component={[Function]}
            path="/spirit/sessions"
          />,
          <Route
            component={[Function]}
            path="/spirit/concerns"
          />,
          <Route
            component={[Function]}
            path="/spirit/sessions"
          />,
          <Route
            component={[Function]}
            path="/spirit/sessions/:session_id"
          />,
          <Switch>
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/sam"
            />
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/sensory"
            />
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/modulation"
            />
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/executive"
            />
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/postural"
            />
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/social"
            />
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/finish"
            />
            <Route
              component={[Function]}
              path="/spirit/sessions/:session_id/view"
            />
            <Route
              component={[Function]}
              path="/spirit/concerns/:concernId/sessions"
            />
            <Route
              component={[Function]}
              path="/spirit/patients/:patient_id"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/spirit/users/:user_id/join"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/spirit/users/:user_id/create"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/spirit/users/:user_id"
            />
            <Route
              component={[Function]}
              path="/spirit/users"
            />
            <Route
              component={[Function]}
              path="/spirit/index"
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/spirit/patients",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/spirit/concerns",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/spirit/sessions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/spirit/concerns",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/spirit/sessions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/spirit/sessions/:session_id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/sam"
              />,
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/sensory"
              />,
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/modulation"
              />,
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/executive"
              />,
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/postural"
              />,
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/social"
              />,
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/finish"
              />,
              <Route
                component={[Function]}
                path="/spirit/sessions/:session_id/view"
              />,
              <Route
                component={[Function]}
                path="/spirit/concerns/:concernId/sessions"
              />,
              <Route
                component={[Function]}
                path="/spirit/patients/:patient_id"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/spirit/users/:user_id/join"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/spirit/users/:user_id/create"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/spirit/users/:user_id"
              />,
              <Route
                component={[Function]}
                path="/spirit/users"
              />,
              <Route
                component={[Function]}
                path="/spirit/index"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/sam",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/sensory",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/modulation",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/executive",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/postural",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/social",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/finish",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/sessions/:session_id/view",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/concerns/:concernId/sessions",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/patients/:patient_id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/spirit/users/:user_id/join",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/spirit/users/:user_id/create",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/spirit/users/:user_id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/users",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/spirit/index",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
